File: Python/Queue/BJAlgo_1158.py
from sys import stdin

N, K = map(int, stdin.readline().split())
memberList = []
result = []
popNum = 0

for i in range(1, (N + 1)):
    memberList.append(i)

while bool(memberList):
    popNum = (popNum + (K - 1)) % len(memberList)
    popElement = memberList.pop(popNum)
    result.append(str(popElement))

print("<{}>".format(", ".join(result)))  # "".join(리스트) 은 list => str 으로 전환하며 사이에 어떤 문자를 넣을지 지정할 수 있음

# - 참고 -
# 시간 초과 났던 기존 코드
# from sys import stdin
#
# N, K = map(int, stdin.readline().split())
# memberList = []
# result = []
# count = 0
#
# for i in range(1, (N + 1)):
#     memberList.append(i)
#
# while bool(memberList):
#     data = memberList[0]
#     del memberList[0]
#
#     if count == (K - 1):
#         count = 0
#         result.append(str(data))
#     else:
#         count += 1
#         memberList.append(data)
#
# print("<{}>".format(", ".join(result)))

File: Java/Queue/BJAlgo_1158.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class BJAlgo_1158 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] NK = br.readLine().split(" ");
        int N = Integer.parseInt(NK[0]);
        int K = Integer.parseInt(NK[1]);

        List memberList = new ArrayList();
        List result = new ArrayList();
        int popNum = 0;

        for(int i = 0; i < N; i++) {
            memberList.add(i + 1);
        }

        while (memberList.size() > 0) {
            popNum = (popNum + (K - 1)) % memberList.size();
            String data = String.valueOf(memberList.get(popNum));
            memberList.remove(popNum);
            result.add(data);
        }

        System.out.println("<" + String.join(", ", result) + ">");
    }
}
